docs/DeveloperGuide.md

```diff
@@ -2,28 +2,206 @@
 
 ## Acknowledgements
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+1. Reference to AB-3 Developer Guide
+
+- [Source](https://se-education.org/addressbook-level3/DeveloperGuide.html#proposed-undoredo-feature)
+- Used as template to structure this Developer Guide
 
 ## Design & implementation
 
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+---
+
+### Ui component
+
+The `Ui` class handles message output for the user within the terminal.
+
+The `UI`component,
+- stores message output strings for user feedback
+- holds methods for printing messages to be called by other classes, such as `InputParsing`.
+
+#### Design considerations
+
+Being a CLI application, UI/UX is minimal and user IO is confined within the terminal.
+
+Hence, public methods within the`Ui` class are static for easy access by other classes, without the need to instantiate an instance of the `Ui` class to call.
+
+### InputParsing component
+
+This component ensures that the user parses in commands in a format that makes sense, which will modify the master list.
+
+![InputParsingUML](./diagrams/src/InputParsing/InputParsing.png)
```

Comment:
Wow, you are using a package diagram! Maybe, the use of a Package Diagram might be confusing?